src/session.rs

```diff
@@ -872,80 +871,29 @@ where
-                            "invalid session id" => ErrorStatus::InvalidSessionId,
-                            "no such element" => ErrorStatus::NoSuchElement,
-                            "no such window" => ErrorStatus::NoSuchWindow,
-                            "no such alert" => ErrorStatus::NoSuchAlert,
-                            "stale element reference" => ErrorStatus::NoSuchElement,
-                            _ => unreachable!(
-                                "received unknown error ({}) for NOT_FOUND status code",
-                                error
-                            ),
-                        },
-                        StatusCode::INTERNAL_SERVER_ERROR => match error {
-                            "javascript error" => ErrorStatus::JavascriptError,
-                            "move target out of bounds" => ErrorStatus::MoveTargetOutOfBounds,
-                            "session not created" => ErrorStatus::SessionNotCreated,
-                            "unable to set cookie" => ErrorStatus::UnableToSetCookie,
-                            "unable to capture screen" => ErrorStatus::UnableToCaptureScreen,
-                            "unexpected alert open" => ErrorStatus::UnexpectedAlertOpen,
-                            "unknown error" => ErrorStatus::UnknownError,
-                            "script timeout" => ErrorStatus::ScriptTimeout,
-                            "unsupported operation" => ErrorStatus::UnsupportedOperation,
-                            "timeout" => ErrorStatus::Timeout,
-                            _ => unreachable!(
-                                "received unknown error ({}) for INTERNAL_SERVER_ERROR status code",
-                                error
-                            ),
-                        },
-                        StatusCode::REQUEST_TIMEOUT => match error {
-                            "timeout" => ErrorStatus::Timeout,
-                            "script timeout" => ErrorStatus::ScriptTimeout,
-                            _ => unreachable!(
-                                "received unknown error ({}) for REQUEST_TIMEOUT status code",
-                                error
-                            ),
-                        },
-                        StatusCode::METHOD_NOT_ALLOWED => match error {
-                            "unknown method" => ErrorStatus::UnknownMethod,
-                            _ => unreachable!(
-                                "received unknown error ({}) for METHOD_NOT_ALLOWED status code",
-                                error
-                            ),
-                        },
-                        _ => unreachable!("received unknown status code: {}", status),
-                    }
+                    body["error"]
+                        .as_str()
+                        .map(ErrorStatus::from)
+                        .unwrap_or(ErrorStatus::UnknownError)
```

Comment:
This change means that we no longer require that the correct HTTP status code is used for each error kind. Is that intentional?

Reply:
Yes. Let me know if you disagree.
The string itself contains all of the information needed to parse the error type.

My reasoning is as follows:
1. It seems to me the only reason to be more strict with the HTTP status code is if we want to test the WebDriver implementation itself, but I doubt most users of this crate would be interested in that.
Anyone wanting to test the veracity of a particular WebDriver implementation should probably have their own test suite rather than relying on crates like this to report an incorrect HTTP status code.
2. As a user, if I happened to be using a WebDriver implementation that had a bug where it returned the correct error but incorrect HTTP status, I'd rather that my code continued to work than be blocked by an otherwise inconsequential upstream bug.

Reply:
Also there is no situation where the same error string should be treated differently depending on the HTTP status code

Reply:
It's that very last sentence I'm concerned about, but given that that's not the case (and we could work around it if it were to be the case), I'm fine with the change :+1: